### PR TITLE
Adding multi node IT

### LINKED ISSUE #354: [CI] Add at least one multi-node integ test
### Is your feature request related to a problem?

Our integration tests only run on a single node cluster, as noted [here](https://github.com/opensearch-project/flow-framework/pull/353#issuecomment-1876045331).

### What solution would you like?

At least one integ test should run on multi-node cluster.

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
adding multi node IT 

### Issues Resolved
resolves #354 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ohltyler** [comment]
CI is not running, is there a syntax error? I believe `run` shouldn't be tabbed in.

**Review by dbwiddis — APPROVED** [review]
LGTM, although I don't think we need to have 7 copies of this!   Could probably put a conditional on the multi-node test to only do for one if the JDKs.

**Comment by owaiskazi19** [comment]
@amitgalitz I see the multi node tests are failing with TimeoutException, do we have any idea about it?

**Comment by dbwiddis** [comment]
> I see the multi node tests are failing with TimeoutException, do we have any idea about it?

Yeah let's get the single node tests passing consistently and see if it fixes these, if so merge away.

**Comment by owaiskazi19** [comment]
> Yeah let's get the single node tests passing consistently and see if it fixes these, if so merge away.

Still fails with the same exception. Do we have to add a [TimeoutSuite](https://javadoc.io/static/com.carrotsearch.randomizedtesting/randomizedtesting-runner/2.1.6/com/carrotsearch/randomizedtesting/annotations/TimeoutSuite.html) for the integ tests?

**Comment by dbwiddis** [comment]
> Still fails with the same exception. Do we have to add a [TimeoutSuite](https://javadoc.io/static/com.carrotsearch.randomizedtesting/randomizedtesting-runner/2.1.6/com/carrotsearch/randomizedtesting/annotations/TimeoutSuite.html) for the integ tests?

Looks to me like tests are running before the cluster is up and connected.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -78,3 +78,7 @@ jobs:
     - name: Build and Run Tests
       run: |
           ./gradlew integTest yamlRestTest
+    - name: Multi Nodes Integration Testing
+      if: matrix.java == 21
```
Comment (owaiskazi19) [inline .github/workflows/CI.yml:82]:
```suggestion
      if: ${{ matrix.java == 21 }}
```

Comment (amitgalitz) [reply .github/workflows/CI.yml:82]: current code works, multi node only ran on jdk21

**Comment by amitgalitz** [comment]
getting lots of time out exceptions locally now too, I didn't see this before, so trying out a few different commits locally to see if I can locate any big differences

File: src/test/java/org/opensearch/flowframework/rest/FlowFrameworkRestApiIT.java [hunk 2]
```diff
@@ -175,15 +175,25 @@ public void testCreateAndProvisionRemoteModelWorkflow() throws Exception {
...
+            () -> { getAndAssertWorkflowStatus(workflowId, State.NOT_STARTED, ProvisioningProgress.NOT_STARTED); },
+            30,
+            TimeUnit.SECONDS
+        );
 
         // Hit Provision API and assert status
         response = provisionWorkflow(workflowId);
         assertEquals(RestStatus.OK, TestHelpers.restStatus(response));
-        getAndAssertWorkflowStatus(workflowId, State.PROVISIONING, ProvisioningProgress.IN_PROGRESS);
+        Thread.sleep(500);
```
Comment (owaiskazi19) [inline src/test/java/org/opensearch/flowframework/rest/FlowFrameworkRestApiIT.java:188]: Can you make this 1000 and try again for mac-os?

**Comment by joshpalis** [comment]
Seems removing the wipreAllIndices method resolved most of the issues we're seeing. All but 1 multi-node test passed

```
1> [2024-01-12T04:01:49,541][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionAgentFrameworkWorkflow] before test
  1> [2024-01-12T04:01:49,544][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionAgentFrameworkWorkflow] initializing REST clients against [http://[::1]:36829, http://127.0.0.1:41505,/ http://[::1]:46719, http://127.0.0.1:41221,/ http://[::1]:41357, http://127.0.0.1:46585]/
  1> [2024-01-12T04:02:27,917][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionAgentFrameworkWorkflow] after test
  1> [2024-01-12T04:02:27,922][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testSearchWorkflows] before test
  1> [2024-01-12T04:02:28,156][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testSearchWorkflows] after test
  1> [2024-01-12T04:02:28,160][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionLocalModelWorkflow] before test
  1> [2024-01-12T04:06:41,908][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionLocalModelWorkflow] after test
  1> [2024-01-12T04:06:41,913][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionRemoteModelWorkflow] before test
  1> [2024-01-12T04:11:16,175][INFO ][o.o.f.r.FlowFrameworkRestApiIT] [testCreateAndProvisionRemoteModelWorkflow] after test

Tests with failures:
 - org.opensearch.flowframework.rest.FlowFrameworkRestApiIT.testCreateAndProvisionRemoteModelWorkflow

5 tests completed, 1 failed
```

**Comment by amitgalitz** [comment]
closing for #416

**Pull request closed without merging**